tock/hours/tests/test_models.py

```diff
@@ -200,13 +200,21 @@ def setUp(self):
             g_start_date=datetime.date(2016, 1, 1)
         )
         self.reporting_period = ReportingPeriod.objects.create(
+            start_date=datetime.date.today() - datetime.timedelta(days=14),
+            end_date=datetime.date.today() - datetime.timedelta(days=7)
+        )
+        self.reporting_period2 = ReportingPeriod.objects.create(
             start_date=datetime.date.today() - datetime.timedelta(days=7),
             end_date=datetime.date.today()
         )
         self.timecard = Timecard.objects.create(
             user=self.user[0],
             reporting_period=self.reporting_period
         )
+        self.timecard2 = Timecard.objects.create(
+            user=self.user[0],
```

Comment:
This may just be me, but I'd modify L195 to always return [0] from get_or_create. I think self.user[0] could read as though there is a user_set, which is just weird. If you need self.user[1], I'd just assign it to another var.

Reply:
Ah okay, I'll try and adjust a bit here for clarity!  And for future reference, in Tock parlance the time period is the reporting period, which must be created first before any timecards can be generated.  The timecards in turn are essentially a container for a reporting period tied to a specific user that contains one or more Tock lines (timecard objects), which are the mappings of hours worked to a project for a user.  Does that help clear up some of the confusion at all?

Reply:
It does. I'd still like to see a better var name than timecard2, if we can think of one, though